app-config-react-native-transformer/src/index.ts

```diff
@@ -0,0 +1,30 @@
+import { loadValidatedConfig } from '@lcdev/app-config';
+import { relative } from 'path';
+import { Transformer, upstreamTransformer } from './upstream-transformer';
+
+/**
+ * Transform is called by Metro for each package resolved in the React Native project
+ */
+export const transform: Transformer['transform'] = ({ src, filename, options }) => {
+  // Get the relative path to app-config's entry point
+  const appConfigPath = relative(
+    options.projectRoot,
+    require.resolve('@lcdev/app-config', {
+      // Search relative to the project's root
+      paths: [options.projectRoot],
+    }),
+  );
+
+  // If the current module isn't app-config, use the default upstream transformer
+  if (filename !== appConfigPath) {
+    return upstreamTransformer.transform({ src, filename, options });
+  }
+
+  // Parse config and overwrite app-config's entry point with exported config JSON
+  return loadValidatedConfig().then(({ parsedNonSecrets }) => {
+    const modifiedSrc = `
+      module.exports = ${JSON.stringify(parsedNonSecrets)};
+    `;
+    return upstreamTransformer.transform({ src: modifiedSrc, filename, options });
+  });
```

Comment:
@joelgallant I've tested this on a React Native project successfully, though environment variables don't seem to parse. This is the config object as seen from inside the RN app:
```
{
  "apiUrl": "http://${MY_IP:-localhost}:3838",
}
```
Did something change with the implementation of `loadValidatedConfig`?

Reply:
Indeed, this feature is now explicit. `{ $substitute: '${MYVAR}' }` is now required. In v1, it's actually not really possible to put a `$` character in strings, which would suck for secrets that we don't generate ourselves.

Reply:
Awesome, thanks. All works as expected 👍